components/vocabularies/orphanet/api/src/main/java/org/phenotips/vocabulary/internal/solr/Orphanet.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.vocabulary.internal.solr;
+
+import org.phenotips.vocabulary.VocabularyTerm;
+
+import org.xwiki.component.annotation.Component;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.common.SolrInputDocument;
+
+import com.hp.hpl.jena.ontology.OntClass;
+import com.hp.hpl.jena.ontology.OntModel;
+import com.hp.hpl.jena.ontology.Restriction;
+import com.hp.hpl.jena.rdf.model.Property;
+import com.hp.hpl.jena.rdf.model.RDFNode;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.Statement;
+
+/**
+ * Provides access to the ORPHANET ontology. The ontology prefix is {@code ORPHANET}.
+ *
+ * @version $Id$
+ * @since 1.2M5
+ */
+@Component
+@Named("orphanet")
+@Singleton
+public class Orphanet extends AbstractOWLSolrVocabulary
+{
+    @Override
+    public String getIdentifier()
+    {
+        return "orphanet";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "Orphanet";
+    }
+
+    @Override
+    protected String getCoreName()
+    {
+        return getIdentifier();
+    }
+
+    @Override
+    protected int getSolrDocsPerBatch()
+    {
+        return 15000;
+    }
+
+    @Override
+    public String getDefaultSourceLocation()
+    {
+        return "http://data.bioontology.org/ontologies/ORDO/submissions/7/download";
+    }
+
+    @Override
+    public VocabularyTerm getTerm(String id)
+    {
+        VocabularyTerm result = super.getTerm(id);
+        if (result == null) {
+            String optionalPrefix = this.getName() + ":";
+            if (StringUtils.startsWith(id, optionalPrefix)) {
+                result = getTerm(StringUtils.substringAfter(id, optionalPrefix));
```

Comment:
Maybe this "second chance" mechanism (trying with id as is, and then without "Orphanet:") should be in a higher level class.